src/mono-runtimes/mono-runtimes.targets

```diff
@@ -570,8 +540,14 @@
     />
   </Target>
 
+  <Target Name="_GetMonoBundleItemsInput">
+    <ItemGroup>
+      <_BundleItemInput Include="$(MonoSourceFullPath)\**\*" />
```

Comment:
NOPE. *NOPE*. **NOPE**. ***NOPE***.

This will *kill* rebuild times and leave me screaming in agony.

Consider this simple project:

```xml
<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <Configuration>Debug</Configuration>
  </PropertyGroup>
  <Import Project="Configuration.props" />
  <Target Name="Build">
    <ItemGroup>
      <_BundleItemInput Include="$(MonoSourceFullPath)\**\*" />
    </ItemGroup>
    <Message Text="Done" />
  </Target>
</Project>
```

Run it:

```bash
$ time msbuild X.targets
...
real	0m12.903s
```

Repeated efforts *do* get faster:

```bash
$ time msbuild X.targets
...
real	0m4.450s
```

but we're still looking at *at least* 4 *seconds* for MSBuild to do that directory traversal, once everything is in the disk cache.

Additionally, `mono-runtimes.csproj` is referenced from *everywhere*.  Rebuild `Mono.Android.csproj`? `mono-runtimes.csproj` is referenced, and thus MSBuild must determine if it must be built (using the `_BuildUnlessCached` target, which hits `_GetMonoBundleItemsInput`).

I *do not* want to turn a target which currently takes *17ms* to execute into one which takes *at least 4 seconds* to execute, simply because you want to enumerate the *96581* files in the mono checkout (give or take), and enumerate that list on *every build*.

NO.

Reply:
Fair enough. The current check is quite broken though.